tpm2/tpm2.go

```diff
@@ -80,28 +79,23 @@ func encodeTPMLPCRSelection(sel PCRSelection) ([]byte, error) {
 	return tpmutil.Pack(uint32(1), ts)
 }
 
-func decodeTPMLPCRSelection(buf []byte) (int, PCRSelection, error) {
-	initialLen := len(buf)
-
+func decodeTPMLPCRSelection(buf *bytes.Buffer) (PCRSelection, error) {
 	var count uint32
 	var sel PCRSelection
-	read, err := tpmutil.Unpack(buf, &count)
-	if err != nil {
-		return 0, sel, err
+	if err := tpmutil.UnpackBuf(buf, &count); err != nil {
+		return sel, err
 	}
-	buf = buf[read:]
 	if count != 1 {
-		return 0, sel, fmt.Errorf("decoding TPML_PCR_SELECTION list longer than 1 is not supported (got length %d)", count)
+		return sel, fmt.Errorf("decoding TPML_PCR_SELECTION list longer than 1 is not supported (got length %d)", count)
 	}
 
 	// See comment in encodeTPMLPCRSelection for details on this format.
 	var ts tpmsPCRSelection
-	read, err = tpmutil.Unpack(buf, &ts.Hash, &ts.Size)
-	if err != nil {
-		return 0, sel, err
+	if err := tpmutil.UnpackBuf(buf, &ts.Hash, &ts.Size); err != nil {
+		return sel, err
 	}
-	buf = buf[read:]
-	ts.PCRs, buf = buf[:ts.Size], buf[ts.Size:]
+	ts.PCRs = make([]byte, ts.Size)
+	buf.Read(ts.PCRs)
```

Comment:
Please check the return value for errors here.

Reply:
Done